src/sdp/media.c

```diff
@@ -54,7 +54,7 @@ static int media_alloc(struct sdp_media **mp, struct list *list)
 	list_append(list, &m->le, m);
 
 	m->ldir  = SDP_SENDRECV;
-	m->rdir  = SDP_SENDRECV;
+	m->rdir  = SDP_INACTIVE;
```

Comment:
this looks like a bad idea. The default value if attribute is missing is SENDRECV.

this SDP stack has been working robust in production environments for 10+ years.
is there really a need to change this now ?

Reply:
> this looks like a bad idea. The default value if attribute is missing is SENDRECV.

And it still is. The sdp_session has its default value SENDRECV. As soon as the m-line is present, it is set in `msg.c:223`
```c
	m->rdir = sess->rdir;
```
If the m-line is missing `m->rdir` will stay at `SDP_INACTIVE`, what is correct.

So the commit should be save.

I very long studied baresip: audio, video, stream and re: sdp/session, sdp/msg, sdp/media. What I found is that everything is allocated for a call: audio/video, streams, sdp_session, sdp_media for audio+video. The SDP decoding only overwrites the default values. But what if the media line is not present in the SDP?

> 
> this SDP stack has been working robust in production environments for 10+ years.
> is there really a need to change this now ?

Only for:
- an audio only setup, where video is not available,
- for audio+video calls

No we have a new feature: audio-only call. So the failure occurs now.
Not working:
For an **audio only call** in an audio+video setup the call is terminated by baresip in the `check_rtp_handler()`.

The commit message should say everything. If not, then let me know!

Reply:
Baresip PR https://github.com/baresip/baresip/pull/1465 makes this one unneeded.